Challo/Challo/Modules/MainContainer/MainContainerPresenter.swift

```diff
@@ -6,18 +6,43 @@
 //
 
 import SwiftUI
+import Combine
 
 class MainContainerPresenter: PresenterProtocol, ObservableObject {
 
+    @Published var profileTab: AnyView?
+    @Published var tabSelection = 0
+
     var interactor: MainContainerInteractor!
     var router: MainContainerRouter?
+    unowned let userState: UserStateProtocol!
+    
+    private var cancellables: Set<AnyCancellable> = []
+
+    init(userState: UserStateProtocol) {
+        self.userState = userState
+        initializeProfileTab()
+    }
+    
+    func initializeProfileTab() {
+        guard let state = userState as? UserState else {
+            return
+        }
+        let loggedInSubscriber = state.$loggedIn.sink(receiveValue: toggleLoggedIn)
```

Comment:
Make sure this does not have a retain cycle! I am pretty sure using the object's function will cause a strong retain cycle. In addition, you can chain the functions to make it nicer.

```swift
self.peggleGameEngine.$gameObjects.sink { [weak self] value in
            guard let self = self else {
                return
            }
            
            self.gameObjectViews = value.map({ self.convertGameObjectToView(gameObject: $0) })
        }.store(in: &cancellables)
```